editor/animation_track_editor.cpp

```diff
@@ -160,6 +160,8 @@ class AnimationTrackKeyEdit : public Object {
 			undo_redo->commit_action();
 
 			setting = false;
+
+			AnimationPlayerEditor::singleton->_animation_key_editor_seek(-1, false);
```

Comment:
`_animation_key_editor_seek(float p_pos, bool p_drag)`

is used like

`_animation_key_editor_seek(-1, false);`

However you're doing a comparison of "-1" which is integer type to a float.

`if (p_pos != -1)`

Maybe you want a boolean?

Reply:
It's not very clear why you're seeking to a valid time, (-1, is a valid time).

Maybe you just want to call.

`EditorNode::get_singleton()->get_inspector()->refresh();`

Reply:
@fire 
I tried replacing : 
`AnimationPlayerEditor::singleton->_animation_key_editor_seek(-1, false);`

With : 

`EditorNode::get_singleton()->get_inspector()->refresh();

As you suggested, It didn't work.

The reason I added -1, is because there is no way to make an animation with values less then 0 as valid time.

So its not a valid time.

Also the reason I didn't add another argument to the function is to not mess with it.

for the `-1` being in `integer `do you suggest me to switch it to `-1.0` ?

Reply:
Let me try modifying the code and try some variations. I think adding an overloaded argument that defaults to false would be better. Will report back.

Reply:
I think the operation you want to do is get the current timeline position, set the value which will set the timeline position to the keyframe timeline position and then seek back to the current timeline position.

Reply:
True.
That's what I want.
Didn't find a connection between `Animation Track Editor` and `Animation Player Editor Plugin`.
Beside `AnimationPlayerEditor::singleton`

So maybe adding a function that return the current timeline position. then feed it back to the seek function.

Reply:
https://github.com/fire/godot/tree/instant-anim @TechnoArtStudio Try this variation.

Reply:
I had a hard time figuring how to use it, this makes a preview but it interpolates, but doesn't save the key in my pr.

Reply:
Feel free to take the code. I'm not able to use it enough to see the problems.

I would recommend you go through each of the animation types and make an example to verify it works.

Reply:
I have done checking your commit.
Your commit is better then mine in which:

-It can be undo and redo, unlike my commit.

So lets go with yours.

You can close this PR and open add your PR.

Reply:
https://github.com/godotengine/godot/pull/43463